### ISSUE TITLE
Can't get Window object for apps with more than one "main" window

### ISSUE THREAD

**Issue by palenshus** [FlaUI/FlaUI#485]
**Describe the bug**
For example, Google Chrome can have multiple open windows, but FlaUI only allows getting the automation controls for the "main" one, which is arbitrary. It'd be great if there was either a GetMainWindow**s** method, or just a FromHandle method like AutomationElement has.

**Code snippets**
```cs
var app = FlaUI.Core.Application.Attach("chrome.exe");
    using (var automation = new UIA3Automation())
    {
        var window = app.GetMainWindow(automation);
```

Currently, I can't use the library, I instead have to enumerate the windows manually (I use the https://github.com/zastrowm/Win32Interop.WinHandles library for this), and use the built-in .NET automation library:

```cs
foreach (var window in EnumerateWindows(chromeProcess))
{
    AutomationElement root = AutomationElement.FromHandle(window);
    ...
}

static IEnumerable<IntPtr> EnumerateWindows(Process process)
{
    return TopLevelWindowUtils.FindWindows(w => w.IsVisible())
        .Where(w =>
        {
            PInvoke.GetWindowThreadProcessId(w.RawPtr, out uint processId);
            return processId == process.Id;
        })
        .Select(w => w.RawPtr);
}
```

**Comment by palenshus** [comment]
@Roemer, what do you think? I'm happy to make this change if you approve

**Comment by kuskitkd** [comment]
Isn't that what app.GetAllTopLevelWindows(automation) does already?

**Comment by palenshus** [comment]
Nope. Maybe it's supposed to, but it currently returns 0 windows for "chrome.exe", whereas GetMainWindow() returns one window. See this LINQPad script for a demo: http://share.linqpad.net/8g76ve.linq

**Comment by dfev77** [comment]
At least on chrome.exe you have multiple processes not just one with multiple windows.
Could this be the reason in your case?

**Comment by palenshus** [comment]
Yes there are multiple processes but only one process has a MainWindowHandle. For an easy repro, open up two chrome Windows and run this LINQPad script: http://share.linqpad.net/q5n2np.linq

In the first line I check that there's only exactly one chrome process with a MainWindowHandle, and then I attempt to get its address bar content using FLaUI, which only processes one window, then I do it again by using UI automation directly, which will get the address bar for both windows.

**Comment by GrizzlyEnglish** [comment]
Just wanted to second this. Moved from WinAppDriver to FlaUI at work and our application (UWP) opens a IE window for logging in the user and the only way capture said window was to utilize the solution posted by @palenshus and utilize the other library.

**Comment by Roemer** [comment]
The `MainWindow` in fact is always only one window at max (this is the MainWindowHandle of the process, which can only be one).
In such a case, you usually use `automation.GetDesktop()` which gives you the absolute root object. There you can search for whatever you want, for example for Windows that belong to and of your chrome processes.

**Comment by palenshus** [comment]
Thanks @Roemer! That's a decent workaround, though I do think it would be better to have a dedicated method. If there's only ever one "MainWindow", then perhaps instead it could be called `GetVisibleWindows()`? In the meantime, I was able to use your suggestion to do the following:

```cs
using var app = FlaUI.Core.Application.Attach(processName);
var visibleWindows = automation.GetDesktop().FindAllChildren(e => e.ByProcessId(app.ProcessId));
```

**Comment by Roemer** [comment]
There is actually a method for that: `GetAllTopLevelWindows` on the `Application` object.

**Comment by palenshus** [comment]
I get different results with that. I tried it using Chrome with two windows open:

```cs
using var app = FlaUI.Core.Application.Attach("chrome.exe");

using var automation = new UIA3Automation();

var topLevelWindows = app.GetAllTopLevelWindows(automation);
topLevelWindows.Count().Dump();    // returns 0

var visibleWindows = automation.GetDesktop().FindAllChildren(e => e.ByProcessId(app.ProcessId));
visibleWindows.Count().Dump();    // returns 2
```

**Comment by StllUnknwn** [comment]
One thing to consider with `GetAllTopLevelWindows` is that it returns only top-level windows, like the main window, not its child windows. To get the windows that are opened by the main window try:
```
mainWindow.FindAllChildren(w => w.ByControlType(FlaUI.Core.Definitions.ControlType.Window))
```
